### ISSUE TITLE
vcpkg: Build failure of 1.12 due to usage of deprecated functions

### ISSUE THREAD

**Issue by uklotzde** [taglib/taglib#1000]
The build of 1.12 for vcpkg fails due to internal invocations of functions that have been marked as deprecated:

https://github.com/microsoft/vcpkg/pull/16266#issuecomment-781152865

**Comment by uklotzde** [comment]
Do you have any ideas how to resolve this issue in the long term?

For the vcpkg build we have included a patch for CMakeLists.txt that simply disabled warning C4996 for the Windows builds, no code changes. Is this the recommended workaround?

**Comment by ufleisch** [comment]
I suppose that the reason for the deprecated functions is to motivate clients of TagLib to no longer use them. If it is a problem when a deprecated function is used internally, its contents could be put into a private function, which is then called by the deprecated function and the recommended successor function.

**Comment by uklotzde** [comment]
This was also my first thought. It would resolve the deprecation warnings.

Ideally the CI build(s) should fail on such warnings as a regression test at build time.

**Comment by sbooth** [comment]
It seems that the internal invocations of deprecated functions occur in functions that are themselves deprecated.

In clang it's possible to push and pop compiler options in the source code. Is this possible with MSVC (or whichever compiler the windows build is using)?

Something like:

```
// This function is deprecated in the header
Frame::Header::Header(const ByteVector &data, bool synchSafeInts) :
  d(new HeaderPrivate())
{
  // This form of setData() is deprecated
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated"
  setData(data, synchSafeInts);
#pragma clang diagnostic pop
}
```

might solve the issue.

**Comment by uklotzde** [comment]
Next attempt to upgrade the vcpkg package to 1.12: https://github.com/microsoft/vcpkg/pull/16649

I would appreciate it if someone from the dev team could verify and confirm the changes and fixes I proposed.

**Comment by ufleisch** [comment]
Fixed by #1062.
